### ISSUE TITLE
Less::Error: Maximum call stack size exceeded with Nanoc >4.3.7

### ISSUE THREAD

**Issue by RubenVerborgh** [nanoc/nanoc#999]
Compiling a simple Less stylesheet is broken in Nanoc >4.3.7.

### Steps to reproduce

1. `git clone https://github.com/LinkedDataFragments/Website.git && cd Website `
2. `git checkout cc9bd7f9f6bd1b5b52ab63eed31eb12705a1fbf8`
3. `bundle install`
4. `bundle exec nanoc`

### Expected behavior

It compiles, as it does with Nanoc 4.3.7.

### Actual behavior

```
Less::Error: Maximum call stack size exceeded
```

### Details

Discovered at LinkedDataFragments/Website#1

### Crash log
[crash.log](https://github.com/nanoc/nanoc/files/608519/crash.log.txt)

**Comment by RubenVerborgh** [comment]
Seeing this issue number, I so badly want to find another bug now.

**Comment by ddfreyne** [comment]
`Maximum call stack size exceeded` is an error from the JavaScript runtime which I’ve started encountering in Nanoc as well, but I have never been able to reproduce it locally (only on Travis CI). I’ve tried to figure out where this is coming from, but to no avail.

**Comment by RubenVerborgh** [comment]
I can reproduce on our Ubuntu server; didn't succeed on OSX. Can give you
access if needed.

**Comment by ddfreyne** [comment]
I’ll see about setting up a Vagrant machine here and testing it out that way!

**Comment by ddfreyne** [comment]
Can you do the following for me:

1. Move the contents of `reset.less` into `main.less`
2. Delete `reset.less`
3. Compile

Does the same issue arise in this case as well?

**Comment by RubenVerborgh** [comment]
Tried that, compilation succeeds!

**Comment by ddfreyne** [comment]
Managed to reproduce the problem. This fails:

```
require 'less'

Fiber.new do
  ::Less::Parser.new.parse("p { color: red }").to_css
end.resume
```

This succeeds:

```
require 'less'

::Less::Parser.new.parse("p { color: red }").to_css
```

It appears that the way fibers work is interfering with V8. Further investigation needed…

**Comment by ddfreyne** [comment]
Reported as cowboyd/less.rb#82.

**Comment by ddfreyne** [comment]
Managed to reduce the error to V8:

```
require 'v8'

context = V8::Context.new

Fiber.new do
  context.eval("console.log('hi')")
end.resume
```

Error:

```
stuff.rb:6:in `block in <main>': Maximum call stack size exceeded (V8::Error)
```

**Comment by ddfreyne** [comment]
Strangely, defining `context` inside the fiber makes that snippet work.

**Comment by ddfreyne** [comment]
Fixed in #1001.

(Admittedly not the cleanest fix, but at least it works!)
